src/cli/net/mod.rs

```diff
@@ -1,12 +1,78 @@
-pub use data::Command;
+// need this to allow MessageEnvelope to have custom serde bounds
+#![allow(clippy::type_repetition_in_bounds)]
 
 mod client;
 mod data;
 mod server;
 
-pub use server::{bind as bind_mpc_helper_server, router as mpc_helper_router, BindTarget};
-
+pub use client::{MpcHandle, MpcHttpConnection as Client};
+pub use data::Command;
 #[cfg(feature = "self-signed-certs")]
 pub use server::tls_config_from_self_signed_cert;
+pub use server::{bind as bind_mpc_helper_server, router as mpc_helper_router, BindTarget};
+use std::str::FromStr;
 
-pub use client::{MpcHandle, MpcHttpConnection as Client};
+use crate::cli::net::server::MpcServerError;
+use crate::protocol::{QueryId, RecordId};
+use async_trait::async_trait;
+use axum::body::Bytes;
+use axum::extract::{FromRequest, RequestParts};
+use axum::http::header::HeaderName;
+
+static OFFSET_HEADER_NAME: HeaderName = HeaderName::from_static("offset");
+static DATA_SIZE_HEADER_NAME: HeaderName = HeaderName::from_static("data-size");
+
+pub struct RecordHeaders {
+    offset: usize,
```

Comment:
what is the valid range for offset? is it possible to use u32 here? in general I am not a big fan of using `usize` for application data because it is platform dependent. It is very rare to actually hit that issue and we could use static assertions to enforce minimum length for usize, but imo it is a good practice to use fixed length integers even if it costs inconveniences while using these as indices

Reply:
ok, I had no strong feelings of using `usize`. i can switch to u32. i would imagine max `offset` == max query size, which probably won't exceed a u32